code/b-parasite/src/prst/adc.c

```diff
@@ -118,5 +127,25 @@ prst_adc_soil_moisture_t prst_adc_soil_read(double battery_voltage) {
                " %% (percentage); %u (relative)",
                ret.raw, NRF_LOG_FLOAT(percentage * 100), ret.relative);
 #endif
+  return ret;
+}
+
+static inline double get_lux_level(double battery_voltage, nrf_saadc_value_t raw_adc_output) {
+  const double x = battery_voltage;
+
+  // todo: add DEBUG lines
```

Comment:
These debug guards are just to help during development. For example, in the `prst_adc_batt_read()` function above, the lines:
```
#if PRST_ADC_BATT_DEBUG
  NRF_LOG_INFO(
      "[adc] Read battery voltage: %d (raw); %d mV; " NRF_LOG_FLOAT_MARKER " V",
      ret.raw, ret.millivolts, NRF_LOG_FLOAT(ret.voltage));
#endif
```

Mean: if `PRST_ADC_BATT_DEBUG` evaluates to true, the `NRF_LOG_INFO` call will be included in the compilation, otherwise it will be stripped out. As a result, when I'm developing and debugging, I set `PRST_ADC_BATT_DEBUG` to `1` [in prst_config.h](https://github.com/rbaron/b-parasite/blob/main/code/b-parasite/config/prst_config.h#L15). Doing so let's me see that `NRF_LOG_INFO` messages on my terminal.

When I'm compiling the final, "production" firmware, I set these debug to `0` to save time & battery, since there's no need to print these values, as nobody is reading them.

Reply:
Excellent. Thanks for the explanation.